foxglove_bridge_base/include/foxglove_bridge/common.hpp

```diff
@@ -74,26 +75,28 @@ struct ClientMessage {
   uint64_t logTime;
   uint64_t publishTime;
   uint32_t sequence;
-  const ClientAdvertisement& advertisement;
+  ClientAdvertisement advertisement;
   size_t dataLength;
-  const uint8_t* data;
+  std::vector<uint8_t> data;
 
   ClientMessage(uint64_t logTime, uint64_t publishTime, uint32_t sequence,
-                const ClientAdvertisement& advertisement, size_t dataLength, const uint8_t* data)
+                const ClientAdvertisement& advertisement, size_t dataLength, const uint8_t* rawData)
       : logTime(logTime)
       , publishTime(publishTime)
       , sequence(sequence)
       , advertisement(advertisement)
```

Comment:
This is now a copy, but I'm assuming `ClientAdvertisement` is relative cheap to copy?

Reply:
Yes I think so too. At this stage I would rather have an extra copy then some hard to find bug. But yes, there is definitely potential for improvement later on